consensus/tendermint/action.go

```diff
@@ -0,0 +1,56 @@
+package tendermint
+
+import (
+	"time"
+)
+
+type Action[V Hashable[H], H Hash, A Addr] interface {
+	Execute(d *Driver[V, H, A])
+}
+
+type BroadcastProposal[V Hashable[H], H Hash, A Addr] Proposal[V, H, A]
+
+type BroadcastPrevote[V Hashable[H], H Hash, A Addr] Prevote[H, A]
+
+type BroadcastPrecommit[V Hashable[H], H Hash, A Addr] Precommit[H, A]
+
+type ScheduleTimeout[V Hashable[H], H Hash, A Addr] timeout
+
+func (a *BroadcastProposal[V, H, A]) Execute(d *Driver[V, H, A]) {
+	d.broadcasters.ProposalBroadcaster.Broadcast(Proposal[V, H, A](*a))
```

Comment:
Is the copy made in this method intentional?

We are re interpreting `a` as a different type and copying it, why not use a reference instead?

Reply:
Is it ok if I address this in another PR? It needs some changes in tests, which will conflict with my next PR.